src/monitor_trigger.py

```diff
@@ -0,0 +1,275 @@
+    """
+    gdf_aoi, iri_prob = setup()
+    iri_prob.download(clobber=True)
+    iri_prob.process(clobber=True)
+    # Load the data that contains the probability per tercile
+    # this allows us to solely look at the below-average tercile
+    # C indicates the tercile (below-average, normal, or above-average).
+    # F indicates the publication month, and L the leadtime
+    ds_iri = iri_prob.load()
+    ds_iri_mask = approx_mask_raster(ds_iri)
+    # only include the area of interest
+    ds_iri_mask = ds_iri_mask.rio.clip(gdf_aoi["geometry"], all_touched=False)
+
+    # compute the statistics on the area
+    df_stats_aoi_bavg = compute_stats_iri(
+        da=ds_iri_mask.prob,
+        gdf_aoi=gdf_aoi,
+        adm0_col="ADM0_FR",
+        pcode0_col="ADM0_PCODE",
+        perc_area=constants.perc_area,
+        threshold_bavg=constants.threshold_bavg,
+        threshold_diff=constants.threshold_diff,
+    )
+
+    # select the months and leadtimes included in the trigger
+    df_stats_aoi_bavg_trig_mom = df_stats_aoi_bavg[
+        df_stats_aoi_bavg[["pub_month", "L"]]
+        .apply(tuple, axis=1)
+        .isin(constants.trig_mom)
+    ]
+
+    # TODO: would it make more sense to have one file per prediction
+    #  date? And possibly even only save the prediction dates that are
+    #  included in the trigger?
+    df_stats_aoi_bavg.to_csv(
+        get_trigger_output_filename(base_dir=iri_prob._processed_base_dir)
+    )
+
+    df_stats_aoi_bavg_trig_mom.to_csv(
+        get_trigger_output_filename(
+            base_dir=iri_prob._processed_base_dir, only_trig_mom=True
+        )
+    )
+
+
+def setup():
+    codab = CodAB(country_config=constants.country_config)
+    gdf_adm1 = codab.load(admin_level=1)
+    gdf_aoi = gdf_adm1[gdf_adm1["ADM1_FR"].isin(constants.adm_sel)]
+    geobb = GeoBoundingBox.from_shape(gdf_adm1)
+    iri_prob = IriForecastProb(
+        country_config=constants.country_config, geo_bounding_box=geobb
+    )
+    return gdf_aoi, iri_prob
+
+
+# TODO: move this to toolbox utils?
+def approx_mask_raster(
+    ds: xr.Dataset,
+    resolution: float = 0.05,
+) -> xr.Dataset:
+    """
+    Resample raster data to given resolution.
+
+    Uses as resample method nearest neighbour, i.e. aims to keep the values
+    the same as the original data. Mainly used to create an approximate mask
+    over an area
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+        Dataset to resample.
+    resolution: float, default = 0.05
+        Resolution in degrees to resample to
+
+    Returns
+    -------
+        Upsampled dataset
+    """
+    upsample_list = []
+    # can only do reproject on 3D array so
+    # loop over all +3D dimensions
+    list_dim = [
+        d for d in ds.dims if (d != ds.aat.x_dim) & (d != ds.aat.y_dim)
+    ]
+    # select from second element of list_dim since can loop over 3D
+    # loop over all combs of dims
+    for i in itertools.product(*[ds[d].values for d in list_dim[1:]]):
```

Comment:
Can you use a more descriptive variable name for `i`, something like, `dim_values` ?